Add plugin: One Step Wiki Link

busyoGG opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:https://github.com/busyoGG/OneStepWikiLink

## Release Checklist
- [X] I have tested the plugin on
  - [X]  Windows
  - [ ]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [X] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [X] `main.js`
  - [X] `manifest.json`
  - [X] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [X] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [X] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [X] I have added a license in the LICENSE file.
- [X] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[const path = require("path");](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L4)
This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[language: Language;](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L15), [language: Language.CN,](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L24), [new Setting(containerEl)](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/setting.ts#L85-L99)
This should not be a setting, use `getLanguage` from the Obsidian API to get the language configured in the app.
Note that this requires the `minAppVersion` to be set to 1.8.0

[autoTimer: NodeJS.Timeout | undefined;](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L54-L55)
Use `number` as the type here and use `window.setTimeout` / `window.clearTimeout` where this is being used.

[name: "Convert All Matching Words to Wiki Links",](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L131)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI) and you should also add this to your i18n configuration.

busyoGG commented:
> [const path = require("path");](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L4) This import is only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.
> 
> [language: Language;](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L15), [language: Language.CN,](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L24), [new Setting(containerEl)](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/setting.ts#L85-L99) This should not be a setting, use `getLanguage` from the Obsidian API to get the language configured in the app. Note that this requires the `minAppVersion` to be set to 1.8.0
> 
> [autoTimer: NodeJS.Timeout | undefined;](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L54-L55) Use `number` as the type here and use `window.setTimeout` / `window.clearTimeout` where this is being used.
> 
> [name: "Convert All Matching Words to Wiki Links",](https://github.com/busyoGG/OneStepWikiLink/blob/bc474b6727562e3bdd5a94bf28e93dabc7bd7f20/src/main.ts#L131) [Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI) and you should also add this to your i18n configuration.

I have removed path, replaced it with manually concatenated strings, and fixed the other mentioned issues.

I added a Chinese command while keeping both language versions of the command, as the English command is more convenient for typing.
